Alliance.Common/GameModes/Story/Conditions/AgentDeathCondition.cs

```diff
@@ -0,0 +1,66 @@
+﻿using Alliance.Common.GameModes.Story.Behaviors;
+using Alliance.Common.GameModes.Story.Utilities;
+using TaleWorlds.Core;
+using TaleWorlds.MountAndBlade;
+
+namespace Alliance.Common.GameModes.Story.Conditions
+{
+	/// <summary>
+	/// Check if agents entered a zone.
+	/// </summary>
+	public class AgentDeathCondition : Condition
+	{
+
+		[ScenarioEditor(label: "Character", tooltip: "ID of the character to check death.")]
+		public string Character;
+		public SideType Side = SideType.All;
+		[ScenarioEditor(label: "Death quota", tooltip: "How many time character need to die to trigger condition.")]
+		public int DeathQuota = 1;
+		[ScenarioEditor(label: "Repeat", tooltip: "If true, it will trigger each time Death quota is reached once again.")]
+		public bool Repeat;
```

Comment:
Convention is to name boolean if "is..."
```suggestion
		public bool isRepeatedOnQuotaReached; // Maybe a better name can be found
```